app/master/cluster_master.py

```diff
@@ -179,7 +179,7 @@ def _disconnect_slave(self, slave):
         """
         # Mark slave dead. We do not remove it from the list of all slaves. We also do not remove it from idle_slaves;
         # that will happen during slave allocation.
-        slave.is_alive = False
+        slave.set_is_alive(False)
```

Comment:
I am concerned here about the scenario where we want to disconnect the slave, even when the slave is still responsive. Because if that is the case, on line 342 below (in this file), the slave._is_alive property will be set to True the next time the slave.is_alive(use_cached=False) method is invoked.

I feel like that scenario shouldn't happen though (where the _disconnect_slave method gets called, and the slave API is still responsive).